### PR TITLE
SONARPY-1533: Ruff report is not using the correct column and row location

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java [hunk 1]
```diff
@@ -81,22 +81,22 @@ private void onResult(JSONObject result) {
...
     JSONObject endLocation = (JSONObject) result.get("end_location");
-    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol);
     issue.endLocationRow = toInteger(endLocation.get("row"));
+    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol, issue.startLocationRow, issue.endLocationRow);
     consumer.accept(issue);
   }
 
   /*
     Ruff returns the col number of the last char + 1.
     In order to properly read the col number we need to return the number of the last char.
```
Comment (jdodinh) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java:91]: Maybe specify that we need to return the `col number`, not just `number`.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java [hunk 2]
```diff
@@ -81,22 +81,22 @@ private void onResult(JSONObject result) {
...
+  private static Integer correctEndLocationCol(Object value, int startLocationCol, int startLocationRow, int endLocationRow) {
     Integer endLocationCol = toInteger(value);
     if (endLocationCol != null) {
-      if (endLocationCol > startLocationCol) {
-        return endLocationCol - 1;
-      } else {
+      if (endLocationRow == startLocationRow && endLocationCol <= startLocationCol) {
         return startLocationCol;
+      } else {
+        return endLocationCol -1 ;
```
Comment (jdodinh) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java:99]: Nitpick: Maybe we should have a space between the `-` and the `1`, i.e.: `endLocationCol - 1`.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java [hunk 3]
```diff
@@ -81,22 +81,22 @@ private void onResult(JSONObject result) {
...
+    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol, issue.startLocationRow, issue.endLocationRow);
     consumer.accept(issue);
   }
 
   /*
     Ruff returns the col number of the last char + 1.
     In order to properly read the col number we need to return the number of the last char.
    */
-  private static Integer correctEndLocationCol(Object value, int startLocationCol) {
+  private static Integer correctEndLocationCol(Object value, int startLocationCol, int startLocationRow, int endLocationRow) {
```
Comment (jdodinh) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java:93]: I assume that in this method we already assume that `startLocationRow <= endLocationRow`? Just making sure, as I'm not entirely clear on this.

**Review by jdodinh — APPROVED** [review]
LGTM! Left some minor comments.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java (hunk 3 above)
Comment (joke1196) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java:93]: Yes so on the same row we can get the end location column which is either bigger or equal to the start location. For example, on an empty line, we get the same end location and start location

**Pull request merged**